python/ray/tune/tuner_config.py

```diff
@@ -0,0 +1,41 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from ray.tune.schedulers import TrialScheduler
+from ray.tune.suggest import Searcher
+from ray.util import PublicAPI
+
+
+@dataclass
+@PublicAPI(stability="beta")
+class TuneAlgoConfig:
+    """Tune specific config, related to how hyperparameter tuning algorithms
+    are set up.
+
+    Args:
+        metric (str): Metric to optimize. This metric should be reported
+            with `tune.report()`. If set, will be passed to the search
+            algorithm and scheduler.
+        mode (str): Must be one of [min, max]. Determines whether objective is
+            minimizing or maximizing the metric attribute. If set, will be
+            passed to the search algorithm and scheduler.
+        search_alg (Searcher): Search algorithm for optimization.
+        scheduler (TrialScheduler): Scheduler for executing
+            the experiment. Choose among FIFO (default), MedianStopping,
+            AsyncHyperBand, HyperBand and PopulationBasedTraining. Refer to
+            ray.tune.schedulers for more options.
+        num_samples (int): Number of times to sample from the
+            hyperparameter space. Defaults to 1. If `grid_search` is
+            provided as an argument, the grid will be repeated
+            `num_samples` of times. If this is -1, (virtually) infinite
+            samples are generated until a stopping condition is met.
+    """
+
+    # Currently this is not at feature parity with `tune.run`, nor should it be.
+    # The goal is to reach a fine balance between API flexibility and conciseness.
+    # We should carefully introduce arguments here instead of just dumping everything.
+    mode: Optional[str] = None
+    metric: Optional[str] = None
+    search_alg: Optional[Searcher] = None
```

Comment:
Should these come with explicit defaults? Right now I think our `search_alg` default is random search, but I'm not sure we're telling anyone anywhere.